instrumentation/opentelemetry-instrumentation-sklearn/src/opentelemetry/instrumentation/sklearn/__init__.py

```diff
@@ -0,0 +1,679 @@
+# Copyright 2020, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The integration with sklearn supports the scikit-learn compatible libraries,
+it can be enabled by using ``SklearnInstrumentor``.
+
+.. sklearn: https://github.com/scikit-learn/scikit-learn
+
+Usage
+-----
+
+Package instrumentation example:
+
+.. code-block:: python
+
+    from opentelemetry.instrumentation.sklearn import SklearnInstrumentor
+
+    # instrument the sklearn library
+    SklearnInstrumentor().instrument()
+
+    # instrument sklearn and other libraries
+    SklearnInstrumentor(packages=["lightgbm", "xgboost"]).instrument()
+
+
+Model intrumentation example:
+
+.. code-block:: python
+
+    from opentelemetry.instrumentation.sklearn import SklearnInstrumentor
+    from sklearn.datasets import load_iris
+    from sklearn.ensemble import RandomForestClassifier
+    from sklearn.model_selection import train_test_split
+    from sklearn.pipeline import Pipeline
+
+    X, y = load_iris(return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y)
+
+    model = Pipeline(
+        [
+            ("class", RandomForestClassifier(n_estimators=10)),
+        ]
+    )
+
+    model.fit(X_train, y_train)
+
+    SklearnInstrumentor().instrument_estimator(model)
```

Comment:
Nit: should `instrument_estimator` be named `instrument_model`?

Reply:
Terminology-wise models _are_ estimators. I could add a method like this:

```python
    def instrument_model(model):
        self.instrument_estimator(model)
```

but that seemed redundant.